File: Arrays&Hashing/128LongestConsecutiveElements.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        
        numSet = set(nums)
        max_length = 0

        for num in numSet:
            if (num-1) not in numSet:
                curr_length = 0
                while (num + curr_length) in numSet:
                    curr_length += 1
                max_length = max(max_length, curr_length)
        return max_length

File: Arrays&Hashing/128LongestConsecutiveElements.cpp
#include<algorithm>
#include<vector>
#include<unordered_set>
class Solution {
public:
    int longestConsecutive(std::vector<int>& nums) {
        std::unordered_set<int> mySet = std::unordered_set<int>(nums.begin(), nums.end());
        int max_len = 0;
        for(const auto& num : mySet) 
        {
            if(mySet.find(num-1) == mySet.end())
            {
                int curr_len = 0;
                while(mySet.find(num + curr_len) != mySet.end())
                {
                    curr_len++;
                }
                max_len = std::max(max_len,curr_len);
            }
        }
        return max_len;
    }
};